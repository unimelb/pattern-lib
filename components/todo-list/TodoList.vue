<template>
  <SectionWrap
    :class="`bg-${bgColor}`"
    class="todo-list">
    <div class="todo-list__container">
      <div class="todo-list__inner">
        <div class="todo-list__figure">
          <FigureWrap
            :caption="false"
            inset>
            <FigureRounded
              :background-image="img"
              :aria-label="ariaLabel" />
          </FigureWrap>
        </div>
        <div class="todo-list__header-txt">
          <h3 class="section-with-rounded-image__heading">
            {{ heading }}
          </h3>
          <p class="section-with-rounded-image__description">
            {{ description }}
          </p>
        </div>
      </div>
      <div class="todo-list__button-cards">
        <div class="todo-list__fill" />
        <ButtonCard
          v-for="data in buttonCardData"
          :key="data.id"
          :top-label="data.topLabel"
          :label="data.label"
          :link="data.link"
          :inverted="data.inverted" />
      </div>
    </div>
  </SectionWrap>
</template>

<script>
import SectionWrap from 'components/section/SectionWrap.vue';
import FigureRounded from 'components/figure/FigureRounded.vue';
import FigureWrap from 'components/figure/FigureWrap.vue';
import ButtonCard from 'components/buttons/ButtonCard.vue';

export default {
  components: {
    SectionWrap,
    FigureRounded,
    FigureWrap,
    ButtonCard,
  },
  props: {
    img: {
      type: String,
      default: '',
      required: true,
    },
    ariaLabel: {
      type: String,
      default: '',
      required: true,
    },
    heading: {
      type: String,
      default: '',
    },
    description: {
      type: String,
      default: '',
    },
    buttonCardData: {
      type: Array,
      default: () => [{}],
      required: true,
    },
    bgColor: {
      type: String,
      default: 'light-blue',
      validator: (value) => ['light-blue', 'alt'].includes(value),
    },
  },
};
</script>
