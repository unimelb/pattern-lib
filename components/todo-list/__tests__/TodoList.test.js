import {
  mount,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import TodoList from '../TodoList.vue';

expect.extend(toHaveNoViolations);

describe('TodoList', () => {
  it('should match snapshot', () => {
    const result = mount(TodoList, {
      propsData: {
        bgColor: 'light-blue',
        img: 'http://lorempixel.com/640/480/people',
        ariaLabel: 'Duis nulla id sint exercitation sint nisi reprehenderit nulla nulla enim consequat id velit aliqua.',
        heading: 'Pariatur cupidatat incididunt.',
        description: 'In mollit adipisicing commodo tempor proident dolore consectetur in culpa. Non esse reprehenderit et occaecat proident nisi. Do nulla elit cupidatat esse non in nulla.',
        buttonCardData: [{
          topLabel: '29th May',
          label: 'Velit quis non anim deserunt anim aliqua enim.',
          link: 'https://www.unimelb.edu.au',
        },
        {
          label: 'Velit quis non anim deserunt anim aliqua enim.',
          link: 'https://www.unimelb.edu.au',
        },
        {
          label: 'Velit quis non anim deserunt anim aliqua enim.',
          link: 'https://www.unimelb.edu.au',
          inverted: true,
        }],
      },
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = mount(TodoList, {
      propsData: {
        bgColor: 'light-blue',
        img: 'http://lorempixel.com/640/480/people',
        ariaLabel: 'Duis nulla id sint exercitation sint nisi reprehenderit nulla nulla enim consequat id velit aliqua.',
        heading: 'Pariatur cupidatat incididunt.',
        description: 'In mollit adipisicing commodo tempor proident dolore consectetur in culpa. Non esse reprehenderit et occaecat proident nisi. Do nulla elit cupidatat esse non in nulla.',
        buttonCardData: [{
          topLabel: '29th May',
          label: 'Velit quis non anim deserunt anim aliqua enim.',
          link: 'https://www.unimelb.edu.au',
        },
        {
          label: 'Velit quis non anim deserunt anim aliqua enim.',
          link: 'https://www.unimelb.edu.au',
        },
        {
          label: 'Velit quis non anim deserunt anim aliqua enim.',
          link: 'https://www.unimelb.edu.au',
          inverted: true,
        }],
      },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
