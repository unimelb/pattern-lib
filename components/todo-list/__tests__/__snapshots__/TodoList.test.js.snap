// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should match snapshot 1`] = `
<div
  class="section todo-list bg-light-blue"
>
  <div
    class="section__inner"
  >
    <div
      class="todo-list__container"
    >
      <div
        class="todo-list__inner"
      >
        <div
          class="todo-list__figure"
        >
          <figure
            class="figure figure--inset"
          >
            <div
              aria-label="Duis nulla id sint exercitation sint nisi reprehenderit nulla nulla enim consequat id velit aliqua."
              class="figure-rounded"
              role="img"
              style="background-image: url(http://lorempixel.com/640/480/people);"
            />
             
            <!---->
          </figure>
        </div>
         
        <div
          class="todo-list__header-txt"
        >
          <h3
            class="section-with-rounded-image__heading"
          >
            
          Pariatur cupidatat incididunt.
        
          </h3>
           
          <p
            class="section-with-rounded-image__description"
          >
            
          In mollit adipisicing commodo tempor proident dolore consectetur in culpa. Non esse reprehenderit et occaecat proident nisi. Do nulla elit cupidatat esse non in nulla.
        
          </p>
        </div>
      </div>
       
      <div
        class="todo-list__button-cards"
      >
        <div
          class="todo-list__fill"
        />
         
        <a
          class="btn-card"
          href="https://www.unimelb.edu.au"
        >
          <div
            class="btn-card__inner bg-white"
          >
            <div
              class="btn-card__top-label"
            >
              29th May
            </div>
             
            <div
              class="btn-card__label"
            >
              Velit quis non anim deserunt anim aliqua enim.
            </div>
             
            <svg
              aria-hidden="true"
              aria-label="arrow-right"
              class="btn-card__icon"
              focusable="false"
              height="22px"
              role="img"
              width="26px"
            >
              <use
                xlink:href="#icon-arrow-right"
              />
               
              <title>
                Arrow-right
              </title>
               
            </svg>
          </div>
        </a>
        <a
          class="btn-card"
          href="https://www.unimelb.edu.au"
        >
          <div
            class="btn-card__inner bg-white"
          >
            <!---->
             
            <div
              class="btn-card__label"
            >
              Velit quis non anim deserunt anim aliqua enim.
            </div>
             
            <svg
              aria-hidden="true"
              aria-label="arrow-right"
              class="btn-card__icon"
              focusable="false"
              height="22px"
              role="img"
              width="26px"
            >
              <use
                xlink:href="#icon-arrow-right"
              />
               
              <title>
                Arrow-right
              </title>
               
            </svg>
          </div>
        </a>
        <a
          class="btn-card"
          href="https://www.unimelb.edu.au"
        >
          <div
            class="btn-card__inner bg-white btn-card--inverted"
          >
            <!---->
             
            <div
              class="btn-card__label"
            >
              Velit quis non anim deserunt anim aliqua enim.
            </div>
             
            <svg
              aria-hidden="true"
              aria-label="arrow-right"
              class="btn-card__icon"
              focusable="false"
              height="22px"
              role="img"
              width="26px"
            >
              <use
                xlink:href="#icon-arrow-right"
              />
               
              <title>
                Arrow-right
              </title>
               
            </svg>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
